diff --git a/arch/arm/boot/dts/overlays/my-driver-overlay.dts b/arch/arm/boot/dts/overlays/my-driver-overlay.dts
new file mode 100644
index 000000000000..878d082741fb
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/my-driver-overlay.dts
@@ -0,0 +1,16 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2711";
+
+	fragment@0 {
+		target-path="/";
+		__overlay__ {
+			my_device: my_device {
+				compatible = "raspberrypi,my_device";
+			};
+		};
+	};
+
+};
